### PR TITLE
Monolog v1

### PR DESCRIPTION
**Pull request by rmelisson**

### CODE REVIEW

**Comment by revolunet** [comment]
Pour la publication npm, un truc pas mal c'est d'utiliser des commits conventionnels ce qui permet de générer automatiquement changelogs et releases numbers avec semantic-release. cf [nos conventions](https://socialgouv.github.io/support/#/./developpement?id=conventions-de-la-fabrique). 

Exemple de workflow GH basique ici: https://github.com/SocialGouv/hasura-permissions-viewer/blob/master/.github/workflows/release.yml - il faut ajouter quelques hints dans le [package.json](https://github.com/SocialGouv/hasura-permissions-viewer/blob/master/package.json#L25-L35)

Le package devra être avec le scope `@socialgouv`, on te fournira un token associé.

File: src/analysis/Covisit.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+// number of covisited links to be added to each content report
+const LINK_LIMIT = 6;
+
+// minimum covisits to consider as actual link
+// 5 means : a minimum of 5 unique visits where two contents were viewed
+const MIN_OCC = 5;
+
+// we provide a configurable function
+const analyseNoConf = (minOcc, linkLimit) => (dataset) => {
+  // FIXME avoid using to array
```
Comment (revolunet) [inline src/analysis/Covisit.js:15]: `Array.from()` ?

File: src/analysis/Covisit.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+// number of covisited links to be added to each content report
+const LINK_LIMIT = 6;
+
+// minimum covisits to consider as actual link
+// 5 means : a minimum of 5 unique visits where two contents were viewed
+const MIN_OCC = 5;
+
+// we provide a configurable function
+const analyseNoConf = (minOcc, linkLimit) => (dataset) => {
```
Comment (revolunet) [inline src/analysis/Covisit.js:14]: `analyseNoConf` doesnt sound configurable :)

File: src/analysis/popularity.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  const noError = (action) =>
+    ![
+      "https://code.travail.gouv.fr/",
+      "https://code.travail.gouv.fr/?xtor=ES-29-[BIE_202_20200130]-20200130-[https://code.travail.gouv.fr/]",
+      "https://code.travail.gouv.fr/droit-du-travail",
+    ].includes(action.url);
+
+  const filteredVisitViews = uniqueViews.where(noError);
+
+  const removeSection = (url) => {
```
Comment (revolunet) [inline src/analysis/popularity.js:36]: removeAnchor sounds more explicit

File: src/queries.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import * as es from "./elastic";
+import * as Suggestion from "./analysis/suggestion";
+import * as Covisit from "./analysis/covisit";
+
+const Queries = (esClient, index) => {
```
Comment (revolunet) [inline src/queries.js:5]: look ma, no `this` !

File: src/analysis/Covisit.js (hunk 2 above)
Comment (rmelisson) [reply src/analysis/Covisit.js:14]: ahah it was just a temporary function, it meant analyse function without the conf (so, the conf can be passed), but yeah at least confusing ;) will change this anyway

File: src/analysis/Covisit.js (hunk 1 above)
Comment (rmelisson) [reply src/analysis/Covisit.js:15]: it's more of a Dataforge issue here, as I'd like to be able manipulate the dataframe without converting it to an actual array

**Pull request merged**